src/spatial_reactionsystem.jl

```diff
@@ -0,0 +1,142 @@
+### Fetch Required Packages ###
+
+# Very WIP (wanted to do with {S,T} and stuff, but got various errors, and wasn't needed right now).
+### Spatial Reaction Structure. ###
+### Describing a spatial reaction that involves species from two neighbouring compartments. ###
+struct SpatialReaction
+    """The rate function (excluding mass action terms)."""
+    rate::Any
+    """Reaction substrates."""
+    substrates::Tuple{Vector,Vector}
+    """Reaction products."""
+    products::Tuple{Vector,Vector}
+
+    """
+    `false` (default) if `rate` should be multiplied by mass action terms to give the rate law.
+    `true` if `rate` represents the full reaction rate law.
+    """
+    only_use_rate::Bool
+
+    function SpatialReaction(rate,substrates,products;only_use_rate=false)
+        new(rate,substrates,products,only_use_rate)
+    end
+    
+end
+
+### Lattice Reaction Network Structure ###
+# Couples:
+# A network of compartments (a meta graph that can contain some additional infro for each compartment).
+# A reaction network (that is simulated within each compartment).
+# A set of spatial reactions (denoting interaction between comaprtments).
+struct LatticeReactionSystem # <: MT.AbstractTimeDependentSystem # Adding this part messes up show, disabling me from creating LRSs
+    """The reaction system expanded onto a graph."""
```

Comment:
Why expand? Why not handle expansions at conversion time if needed? Jumps shouldn't need expansion, and  at some point ModelingToolkit might get a graph ODE system type we could then exploint.

Reply:
You mean when converting to ODE/SDE/Jump  systems? Yes, that might be better.

Reply:
Yes. For example, with jumps we don't want to flatten. That is the whole point of the spatial jump solvers (which are much faster and more memory efficient than flattening).